optaplanner-integration/optaplanner-spring-boot-example/src/main/java/org/optaplanner/spring/boot/example/service/TimeTableController.java

```diff
@@ -39,117 +29,36 @@
 @RequestMapping("/timeTable")
 public class TimeTableController {
 
-    @Autowired
-    SolverManager<TimeTable> solverManager;
+    private final TimetableService timetableService;
 
-    private AtomicReference<TimeTable> timeTableReference = new AtomicReference<>(generateProblem());
-    private SolverFuture solverFuture = null;
+    public TimeTableController(TimetableService timetableService) {
+        this.timetableService = timetableService;
+    }
 
     // To try, open http://localhost:8080/timeTable
     @RequestMapping()
     public TimeTable refreshTimeTable() {
-        return timeTableReference.get();
+        return timetableService.refreshTimeTable();
     }
 
     @PostMapping("/solve")
     public void solve() {
-        TimeTable timeTable = timeTableReference.get();
-        // TODO Race condition if room is added while solving, it disappears
-        solverFuture = solverManager.solve(timeTable, timeTableReference::set);
+        timetableService.solve();
     }
 
     @PostMapping("/addLesson")
     public void addLesson(@RequestBody Lesson lesson) {
-        timeTableReference.updateAndGet(timeTable -> {
-            long nextId = timeTable.getLessonList().stream()
-                    .map(Lesson::getId).max(Comparator.naturalOrder())
-                    .orElse(0L)
-                    + 1L;
-            timeTable.getLessonList().add(new Lesson(nextId,
-                    lesson.getSubject(), lesson.getTeacher(), lesson.getStudentGroup()));
-            return timeTable;
-        });
+        timetableService.addLesson(lesson);
```

Comment:
these CRUD methods can dissappear if the RoomRepository are exposed as rest services